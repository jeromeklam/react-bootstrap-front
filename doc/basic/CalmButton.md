## CalmButton

LibellÃ© dans une ligne de saisie
