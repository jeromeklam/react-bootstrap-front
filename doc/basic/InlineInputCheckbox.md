## InlineInputCheckbox

Champ interrupteur classique à l'intérieur d'une ligne de saisie.

### Props

#### Obligatoires

| Nom     | Type    | Commentaire                                      |
| ------- | ------- | ------------------------------------------------ |
| name    | Chaine  | Le nom du champ, en snakeCase si possible        |

#### Optionnels

| Nom      | Type    | Remarque | Commentaire                       |
| -------- | ------- | -------- | --------------------------------- |
| id       | Chaine  |          | Identifiant                       |
| label    | Chaine  |          | Le libellé du champ, sans html    |
| disabled | Booléen | Faux     | Pour désactiver le champ          |
| required | Booléen | Faux     | Pour rendre le champ obligatoire  |
| size     | Taille  |          | Pour changer la taille par défaut |
| checked  | Booléen | Faux     | Valeur                            |

#### Fonctions appelées

| Nom          | Type     | Remarque | Commentaire                                       |
| ------------ | -------- | -------- | ------------------------------------------------- |
| onChange     | Fonction | Obl.     | onChange(event)                                   |


### Spécificités

- Le libellé sera écrit à droite de la coche

### Exemple

```
<InlineInputCheckbox
  label="Envoyer les news"
  id="send_news"
  name="send_news"
  labelTop={false}
  size="sm"
  checked={values.send_news}
  onChange={handleChange}
/>
```

### To Do

- Ajouter et gérer l'attribut 'className'
- Ajouter et gérer les attributs 'error' et 'warning'
