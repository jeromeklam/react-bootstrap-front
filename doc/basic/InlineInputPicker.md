## InlineInputPicker

Champ de saisie avec sélection dans une liste de résultats à l'intérieur d'une ligne de saisie.

### Props

#### Obligatoires

| Nom       | Type     | Commentaire                                   |
| --------- | -------- | --------------------------------------------- |
| name      | Chaine   | Le nom du champ, en snakeCase si possible     |
| label     | Chaine   | Le libellé du champ                           |
| list      | Element  | Liste de résultats trouvée avec la saisie     |
| onClear   | Fonction | Action qui vide le champ                      |
| onMore    | Fonction | Action qui ouvre la fenêtre de recherche      |
| onChange  | Fonction | onChange(event)                               |
| onSelect  | Fonction | Pour remplir le champ avec une valeur trouvée |
| clearIcon | Element  | Pour vider le champ                           |
| moreIcon  | Element  | Pour rechercher un élément                    |

#### Optionnels

| Nom           | Type    | Remarque | Commentaire                                                          |
| ------------- | ------- | -------- | -------------------------------------------------------------------- |
| value         | Chaine  | ''       | La valeur                                                            |
| required      | Booléen | Faux     | Pour rendre le champ obligatoire                                     |
| labelTop      | Booléen | Vrai     | Le libellé s'affiche au dessus du champ                              |
| size          | Taille  |          | Pour changer la taille par défaut                                    |
| inputSize     | Taille  | 36       | Taille du champ de saisie                                            |
| error         | Element | Faux     | Gestion du champ en erreur                                           |
| display       | Chaine  | ''       | Chaine cherchée                                                      |
| pickerId      | Chaine  | ''       | Nom de l'identifiant de la table dans laquelle on effectue lacherche |
| pickerDisplay | Chaine  | ''       | Liste des champs (séparés par une virgule) à afficher                |

#### Fonctions appelées

| Nom      | Type     | Remarque | Commentaire                                             |
| -------- | -------- | -------- | ------------------------------------------------------- |
| onClear  | Fonction | Obl.     | Action qui vide le champ                                |
| onMore   | Fonction | Obl.     | Action qui ouvre la fenêtre de recherche                |
| onChange | Fonction | Obl.     | onChange(event)                                         |
| onSelect | Fonction | Obl.     | Pour remplir le champ avec une valeur trouvée           |

### Spécificités

- Si 'error' est renseigné, le champ sera invalide ![](./is_invalid.png) et
  l'élement erreur anomalie sera ajouté après
  (le plus souvent celà sera un message d'erreur en dessous)    
- 2 fonctionnements :
  - Soit on saisit une valeur et si des éléments correspondent,
  une liste s'ouvre dans laquelle on peut sélectionné l'élement voulu
  - Soit, on clique directement sur le bouton recherche (onMore) qui ouvre une fenêtre
  qu'il faut avoir créée avec des champs de recherche qui donne aussi une liste de résultats
-Pas de zoom ni de création à la volée

### Exemple

Exemple non trouvé
