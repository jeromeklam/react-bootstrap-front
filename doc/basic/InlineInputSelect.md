## InlineInputSelect

Zone de saisie avec liste déroulante à l'intérieur d'une ligne de saisie.

### Props

#### Obligatoires

| Nom          | Type     | Commentaire                                       |
| ------------ | -------- | ------------------------------------------------- |
| name         | Chaine   | Le nom du champ, en snakeCase si possible         |

#### Optionnels

| Nom      | Type    | Remarque | Commentaire                                 |
| -------- | ------- | -------- | ------------------------------------------- |
| id       | Chaine  |          | Identifiant                                 |
| value    | Chaine  |          | La valeur                                   |
| disabled | Booléen | Faux     | Pour désactiver le champ                    |
| required | Booléen | Faux     | Pour rendre le champ obligatoire            |
| size     | Taille  |          | Pour changer la taille par défaut           |
| error    | Element |          | Gestion du champ en erreur                  |
| options  | Element | []       | Liste des options de la liste déroulante    |
| addEmpty | Booléen | Faux     | Ajout d'une valeur vide en plus des options |

#### Fonctions appelées

| Nom          | Type     | Remarque | Commentaire                                       |
| ------------ | -------- | -------- | ------------------------------------------------- |
| onChange     | Fonction |          | onChange(event)                                   |

### Spécificités

- Si 'id' non renseigné, l'identifiant sera 'name'
- Si 'error' est renseigné l'élement erreur sera ajouté après

### Exemple

```
<InlineInputSelect
  name="lang.id"
  size="sm"  
  inputSize={36}
  value={values.lang.id}
  onChange={handleChange}
  options={langAsOptions(props.languages)}
  error={getErrorMessage('lang.id')}
/>
```

### To Do

- Ajouter et gérer l'attribut className ?
- Ajouter la gestion d'une valeur par défaut addDefaultValue (comme InputSelect) au lieu du addEmpty?
- Gérer l'id comme l'InputText avec un nombre aléatoire ajouter à 'name'
