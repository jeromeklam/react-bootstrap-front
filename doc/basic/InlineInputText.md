## InputText

Champ de saisie texte à l'intétieur d'une ligne de saisie.

### Props

#### Obligatoires

| Nom          | Type     | Commentaire                                       |
| ------------ | -------- | ------------------------------------------------- |
| name         | Chaine   | Le nom du champ, en snakeCase si possible         |

#### Optionnels

| Nom          | Type     | Remarque | Commentaire                                       |
| ------------ | -------- | -------- | ------------------------------------------------- |
| id           | Chaine   |          | Identifiant                                       |
| value        | Chaine   |          | La valeur                                         |
| disabled     | Booléen  | Faux     | Pour désactiver le champ                          |
| required     | Booléen  | Faux     | Pour rendre le champ obligatoire                  |
| size         | Taille   |          | Pour changer la taille par défaut                 |
| error        | Element  |          | Gestion du champ en erreur                        |
| placeholder  | Chaine   | ''       | Texte indicatif dans le champ de saisie           |
| pattern      | Chaine   | ''       | Modèle de saisie (expression régulière) !         |

#### Fonctions appelées

| Nom          | Type     | Remarque | Commentaire                                       |
| ------------ | -------- | -------- | ------------------------------------------------- |
| onChange     | Fonction |          | onChange(event)                                   |

### Spécificités

- Il n'y a pas de label
- Si 'error' est renseigné l'élement erreur sera ajouté après
- l'attribut 'pattern' peut aider mais tous les navigateurs ne le gère pas
  (https://www.w3schools.com/tags/att_input_pattern.asp),
  il vaut mieux passer pas l'objet inputMask

### Exemple

```
<InlineInputText
  id="ct_code"
  name="ct_code"
  size="sm"
  value={values.ct_code}
  onChange={handleChange}
  error={getErrorMessage('ct_code')}
/>
```

### To Do

- Ajouter et gérer l'attribut 'warning' comme 'error' ?
- Ajouter et gérer l'attribut className ?
