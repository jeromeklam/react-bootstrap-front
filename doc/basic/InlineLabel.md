## InputLabel

Libellé dans une ligne de saisie

### Props

#### Obligatoires

| Nom   | Type   | Commentaire |
| ----- | ------ | ----------- |
| label | Chaine | Libellé     |

#### Optionnels

| Nom     | Type   | Remarque | Commentaire              |
| ------- | ------ | -------- | ------------------------ |
| size    | Chaine | null     | Taille                   |
| htmlFor | Chaine |          | Identifiant du champ lié |

### Exemple

```
<InputLabel
  label="Nom"
/>
```

### To Do

- Ajouter et gérer l'attribut className ?
