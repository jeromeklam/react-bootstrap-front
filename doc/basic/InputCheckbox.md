## InputCheckbox

Champ interrupteur classique (case à cocher).

### Props

#### Obligatoires

| Nom     | Type    | Commentaire                                      |
| ------- | ------- | ------------------------------------------------ |
| name    | Chaine  | Le nom du champ, en snakeCase si possible        |

#### Optionnels

| Nom       | Type    | Remarque | Commentaire                                   |
| --------- | ------- | -------- | --------------------------------------------- |
| id        | Chaine  |          | Identifiant                                   |
| label     | Chaine  |          | Le libellé du champ, sans html                |
| disabled  | Booléen | Faux     | Pour désactiver le champ                      |
| required  | Booléen | Faux     | Pour rendre le champ obligatoire              |
| labelTop  | Booléen | Vrai     | Le libellé s'affiche au dessus du champ       |
| labelSize | Taille  | 6        | Taille du libellé                             |
| inputSize | Taille  | 30       | Taille du champ de saisie                     |
| checked   | Booléen | Faux     | Valeur                                        |
| detail    | Elément | ''       | Elément à mettre après (texte de la checkbox) |

#### Fonctions appelées

| Nom          | Type     | Remarque | Commentaire                                       |
| ------------ | -------- | -------- | ------------------------------------------------- |
| onChange     | Fonction | Obl.     | onChange(event)                                   |

### Spécificités

- Le libellé sera écrit au dessus si labelTop = Vrai sinon il sera écrit avant la coche
  mais pour avoir le texte à droite de la coche, il faut utiliser l'attribut 'detail'

### Exemple

```
<InputCheckbox
  label="Actif"
  name="ct_active"
  labelTop={true}
  checked={values.ct_active === true}
  onChange={handleChange}
/>
```

### To Do

- Ajouter et gérer l'attribut 'className'
