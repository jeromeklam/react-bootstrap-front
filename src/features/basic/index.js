export { default as Dropdown } from "./Dropdown";
export * from "./InlineInputCheckbox";
export * from "./InlineInputPicker";
export { default as InlineInputSelect } from "./InlineInputSelect";
export * from "./InlineInputText";
export * from "./InlineLabel";
export * from "./InputCheckbox";
export { default as InputDateWeb } from "./InputDateWeb";
export { default as InputDatetime } from "./InputDatetime";
export * from "./InputEmail";
export * from "./InputHidden";
export { default as InputMask } from "./InputMask";
export { default as InputMonetary } from "./InputMonetary";
export * from "./InputPassword";
export { default as InputPicker } from "./InputPicker";
export { default as ButtonPicker } from "./ButtonPicker";
export * from "./InputRadio";
export * from "./InputRandomText";
export { default as InputSelect } from "./InputSelect";
export { default as InputSpin } from "./InputSpin";
export { default as InputStringarray } from "./InputStringarray";
export * from "./InputText";
export { default as InputTextarea } from "./InputTextarea";
export * from "./ResponsiveButton";
export { default as InputCheckList } from "./InputCheckList";
export { default as CalmButton } from "./CalmButton";
export { default as InputGroup } from "./InputGroup";
export { default as InputGroupAppend } from "./InputGroupAppend";
export { default as InputGroupPrepend } from "./InputGroupPrepend";
export { default as InputGroupText } from "./InputGroupText";
export { default as DropdownMenu } from "./DropdownMenu";
export { default as DropdownMenuHeader } from "./DropdownMenuHeader";
export { default as DropdownMenuOption } from "./DropdownMenuOption";
export { default as DropdownMenuDivider } from "./DropdownMenuDivider";
export { default as InputKeyword } from "./InputKeyword";
export { default as InputKeywords } from "./InputKeywords";
export { default as InputWorkload } from "./InputWorkload";
export { default as InputGpsCoords } from "./InputGpsCoords";
export { default as InputPhone } from "./InputPhone";
export { default as Badge } from "./Badge";
export { default as InputDate } from "./InputDate";
export * from "./InputDateNatif";
export { default as InputPhoto } from "./InputPhoto";
export { default as InputBarCode } from "./InputBarCode";
export { default as Scanner } from "./Scanner";
export { default as InputSignature } from "./InputSignature";
export * from "./InputTextMultiline";
export * from "./InputChildren";
