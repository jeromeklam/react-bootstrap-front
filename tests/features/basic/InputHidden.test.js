import React from 'react';
import renderer from 'react-test-renderer';
import { InputHidden } from '../../../src/features/basic';

it('renders standard', () => {
  // First render
  const component = renderer.create(
    <InputHidden className="btn bg-secondary" id="input-hidden" name="hidden" value="test" />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
