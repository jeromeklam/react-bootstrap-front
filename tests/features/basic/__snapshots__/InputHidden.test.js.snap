// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders standard 1`] = `
<div>
  <input
    id="input-hidden"
    name="hidden"
    type="hidden"
    value="test"
  />
</div>
`;
